pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -144,25 +150,33 @@ func (m *MachineScope) Close() {
 		m.Error(err, "failed to create new JSONPatch for machine")
 		return
 	}
-	pb, err := json.MarshalIndent(p, "", "  ")
-	if err != nil {
-		m.Error(err, "failed to json marshal patch for machine")
-		return
+
+	if len(p) != 0 {
+		pb, err := json.MarshalIndent(p, "", "  ")
+		if err != nil {
+			m.Error(err, "failed to json marshal patch for machine")
+			return
+		}
+
+		result, err := m.MachineClient.Patch(m.Machine.Name, types.JSONPatchType, pb)
+		if err != nil {
+			m.Error(err, "failed to patch machine")
+			return
+		}
+		// Keep the resource version updated so the status update can succeed
+		m.Machine.ResourceVersion = result.ResourceVersion
 	}
 
-	updated, err := m.MachineClient.Patch(m.Machine.Name, types.JSONPatchType, pb)
-	if err != nil {
-		m.Error(err, "failed to patch machine")
+	// Do not update status if the statuses are the same
+	if reflect.DeepEqual(m.MachineStatus, oldStatus) {
 		return
 	}
 
-	updated.Status.ProviderStatus = status
-	if _, err := m.MachineClient.UpdateStatus(updated); err != nil {
+	m.Machine.Status.ProviderStatus = newStatus
+	if _, err := m.MachineClient.UpdateStatus(m.Machine); err != nil {
 		m.Error(err, "failed to update cluster status")
 		return
 	}
-
-	m.V(1).Info("Updated machine")
```

Comment:
This looks to remove the only positive logging we had for a machine being updated, do we need to log when we've patched the machine or updated the status?

Reply:
I can add this back in. It could be a no-op now though, so a more accurate message might be "scope closed" which I was thinking is not very useful.

Do you think it would be useful to log when we patch and when we update status?

Reply:
+1 to logging when we patch or update status, even if it is a higher verbosity level.

I don't see any reason to log closing the scope.

Reply:
I think https://github.com/kubernetes-sigs/controller-runtime/blob/master/TMP-LOGGING.md this post makes a good argument about Verbose logging:

> While it's possible to use higher log levels, it's recommended that you stick with V(1) or V(0)(which is equivalent to not specifyingV), and then filter later based on key-value pairs or messages; different numbers tend to lose meaning easily over time, and you'll be left wondering why particular logs lines are at V(5)instead ofV(7)`.

I'll add patch & status update as a .V(1)